examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -0,0 +1,139 @@
+# License: BSD (3-clause)
+
+
+import matplotlib.pyplot as plt
+import mne
+from mne import Epochs
+from mne.datasets.fieldtrip_cmc import data_path
+from mne.decoding import SSD
+from mne.utils import _time_mask
+
+###############################################################################
+# Define parameters
+fname = data_path() + '/SubjectCMC.ds'
+raw = mne.io.read_raw_ctf(fname)
+raw.crop(50., 110.).load_data()  # crop for memory purposes
+raw.resample(sfreq=250)
+
+raw.pick_types(meg=True, eeg=False, ref_meg=False)
+
+freqs_sig = 9, 12
+freqs_noise = 8, 13
+
+# Prepare data
+
+ssd = SSD(info=raw.info,
+          sort_by_spectral_ratio=False,  # True is recommended here.
+          filt_params_signal=dict(l_freq=freqs_sig[0], h_freq=freqs_sig[1],
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin'),
+          filt_params_noise=dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin'))
+ssd.fit(X=raw.get_data())
+
+
+###############################################################################
+# Let's investigate spatial filter with max power ratio.
+# We will first inspect the topographies.
+# According to Nikulin et al. 2011 this is done by either inverting the filters
+# (W^{-1}) or by multiplying the noise cov with the filters Eq. (22) (C_n W)^t.
+# We rely on the inversion approach here.
+
+pattern = mne.EvokedArray(data=ssd.patterns_[:4].T,
+                          info=ssd.info)
+pattern.plot_topomap(units=dict(mag='A.U.'), time_format='')
+
+# The topographies suggest that we picked up a parietal alpha generator.
+
+# Transform
+ssd_sources = ssd.transform(X=raw.get_data())
+
+# Get psd of SSD-filtered signals.
+psd, freqs = mne.time_frequency.psd_array_welch(
+    ssd_sources, sfreq=raw.info['sfreq'], n_fft=4096)
+
+# Get spec_ratio information (already sorted).
+spec_ratio, sorter = ssd.get_spectral_ratio(ssd_sources)
+
+# Plot spectral ratio (see Eq. 24 in Nikulin 2011).
+fig, ax = plt.subplots(1)
+ax.plot(spec_ratio, color='black')
+ax.plot(spec_ratio[sorter], color='orange', label='sorted eigenvalues')
+ax.set_xlabel("Eigenvalue Index")
+ax.set_ylabel(r"Spectral Ratio $\frac{P_f}{P_{sf}}$")
+ax.legend()
+ax.axhline(1, linestyle='--')
+
+# We can see that the initial sorting based on the eigenvalues
+# was already quite good. However, when using few components only
+# The sorting mighte make a difference.
+
+###############################################################################
+# Let's also look at the power spectrum of that source and compare it to
+# to the power spectrum of the source with lowest SNR.
+
+below50 = _time_mask(freqs, 0, 50)  # Convenience way for selecting freqs.
+# for highlighting the freq. band of interest
+bandfilt = _time_mask(freqs, freqs_sig[0], freqs_sig[1])
```

Comment:
bandfilt = (freqs_sig[0] <= freqs) & (freqs <= freqs_sig[1])

Reply:
I remember we introduced this private function years ago as this standard notation sometimes led to bugs. But ok to do something more raw and explicit here. I feel we would need a public safe float range selector for either time/freqs .... ```mne.select_times```, ```mne.select_freqs``` .... but not volunteering here :)

Reply:
@vpeterson I'm ok with the suggestions by @larsoner

Reply:
Yes that's why we have it internally but I'd rather not expose it publicly for now